ATOMIC_VAR_INIT shouldn't be used in GC code

On Windows arm64 (it's a VM) I have a Clang 17 installed:

```
$ clang --version
clang version 17.0.6
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: C:\Program Files\LLVM\bin
```

And trying to compile native code fails:

```
[warn] Failed to resolve Scala sources for NIR symbols defined in C:\Users\velvetbaldmime\AppData\Local\Coursier\Cache\v1\https\repo1.maven.org\maven2\org\scala-native\javalib_native0.5_3\0.5.11\javalib_native0.5_3-0.5.11.jar - they would be unavailable in debugger. You can try to add custom custom source directory or jars to config and try again.
[error] C:/Users/velvetbaldmime/projects/cue4s/modules/example/target/native-3/native/dependencies/nativelib_native0.5_3-0.5.11-1/scala-native/gc/immix_commix/GCRoots.c:13:19: error: call to undeclared function 'ATOMIC_VAR_INIT'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
[error]    13 |     roots->head = ATOMIC_VAR_INIT(NULL);
[error]       |                   ^
[error] C:/Users/velvetbaldmime/projects/cue4s/modules/example/target/native-3/native/dependencies/nativelib_native0.5_3-0.5.11-1/scala-native/gc/immix_commix/GCRoots.c:13:17: error: incompatible integer to pointer conversion assigning to 'GC_Root *' (aka 'struct GC_Root *') from 'int' [-Wint-conversion]
[error]    13 |     roots->head = ATOMIC_VAR_INIT(NULL);
[error]       |                 ^ ~~~~~~~~~~~~~~~~~~~~~
[error] 2 errors generated.
```

This macro is deprecated in C17 and removed in C23: https://en.cppreference.com/c/atomic/ATOMIC_VAR_INIT

> This macro was a part of early draft design for C11 atomic types. It is not needed in C11, and is deprecated in C17 and removed in C23. 

I traced the includes to a file `vcruntime_c11_stdatomic.h` and it doesn't have this macro at all.

Seems like it can just be removed.

--- a/nativelib/src/main/resources/scala-native/gc/immix_commix/GCRoots.c
+++ b/nativelib/src/main/resources/scala-native/gc/immix_commix/GCRoots.c
@@ -5,12 +5,11 @@
 #include <stdio.h>
 #include <assert.h>
 #include <stdlib.h>
-#include <stdatomic.h>
 #include "shared/ThreadUtil.h"
 
 GC_Roots *GC_Roots_Init() {
     GC_Roots *roots = (GC_Roots *)malloc(sizeof(GC_Roots));
-    roots->head = ATOMIC_VAR_INIT(NULL);
+    roots->head = NULL;
     mutex_init(&roots->modificationLock);
     return roots;
 }

